### PR TITLE
v1/product: add products gunk definition

### PR DESCRIPTION
**Pull request by jwangsadinata**
This commit adds the products, product attributes and product
collections gunk definition.

### CODE REVIEW

**Comment by jwangsadinata** [comment]
This has been tested to work with the doc-portal site locally.

File: v1/product/product.gunk [hunk 1]
```diff
@@ -0,0 +1,468 @@
...
+//                 "500": openapiv2.Response{
+//                         Description: "Returned when an unexpected error occured on the server side.",
+//                 },
+//         },
+//         SecurityDefinitions: openapiv2.SecurityDefinitions{
+//                 Security: map[string]openapiv2.SecurityScheme{
+//                         "OAuth2": openapiv2.SecurityScheme{
+//                                 Type:             openapiv2.TYPE_OAUTH2,
+//                                 Flow:             openapiv2.FLOW_ACCESS_CODE,
+//                                 Description:      "Access to accounts resources is restricted to authenticated users. The authentication process follows the Authorization Code flow.",
```
Comment (ghazninattarshah) [inline v1/product/product.gunk:39]: change `accounts` to `product`

**Review by nkristianto — APPROVED** [review]
LGTM

File: v1/productcollection/productcollection.gunk [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+//                 "500": openapiv2.Response{
+//                         Description: "Returned when an unexpected error occured on the server side.",
+//                 },
+//         },
+//         SecurityDefinitions: openapiv2.SecurityDefinitions{
+//                 Security: map[string]openapiv2.SecurityScheme{
+//                         "OAuth2": openapiv2.SecurityScheme{
+//                                 Type:             openapiv2.TYPE_OAUTH2,
+//                                 Flow:             openapiv2.FLOW_ACCESS_CODE,
+//                                 Description:      "Access to accounts resources is restricted to authenticated users. The authentication process follows the Authorization Code flow.",
```
Comment (ghazninattarshah) [inline v1/productcollection/productcollection.gunk:39]: change to `product collection`

**Review by ghazninattarshah — COMMENTED** [review]
LGTM with minor change.

**Pull request merged**